lib/babe/verify.go

```diff
@@ -354,17 +354,40 @@ func (b *verifier) verifyAuthorshipRight(header *types.Header) error {
 			continue
 		}
 
+		currentPreDigestItem := currentHeader.Digest.Types[0]
+
+		currentPreDigest, ok := currentPreDigestItem.Value().(types.PreRuntimeDigest)
+		if !ok {
+			return fmt.Errorf("first digest item is not pre-digest")
```

Comment:
I would suggest to create the error `ErrNoFirstPreDigest` and set in the error message the type of `currentPreDigestItem.Value()`
```suggestion
			return fmt.Errorf("%w: got %T", ErrNoFirstPreDigest, currentPreDigestItem.Value())
```